### PR TITLE
Moved assistance to winpr crypto.

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot come on, test!

**Comment by akallabeth** [comment]
@freerdp-bot stop mocking me, test!

**Comment by hardening** [comment]
LGTM, +1

File: libfreerdp/common/assistance.c [hunk 1]
```diff
@@ -95,21 +91,29 @@ int freerdp_assistance_crypt_derive_key_sha1(BYTE* hash, int hashLength, BYTE* k
...
-		return -1;
-
-	SHA1_Init(&hashCtx);
-	SHA1_Update(&hashCtx, pad1, 64);
-	SHA1_Final((void*) buffer, &hashCtx);
-
-	SHA1_Init(&hashCtx);
-	SHA1_Update(&hashCtx, pad2, 64);
-	SHA1_Final((void*) &buffer[hashLength], &hashCtx);
+		goto fail;
```
Comment (bmiklautz) [inline libfreerdp/common/assistance.c]: Just a remark I think the jump here isn't necessary. A return rc would be good as well at this point.

**Comment by bmiklautz** [comment]
+1

**Pull request merged**